src/createAnimatedComponent.js

```diff
@@ -174,19 +175,30 @@ export default function createAnimatedComponent(Component) {
       }
     }
 
+    shouldComponentUpdate(nextProps) {
+      this._attachProps(nextProps);
+      return true;
+    }
+
     componentDidUpdate(prevProps) {
-      this._attachProps(this.props);
       this._reattachNativeEvents(prevProps);
+      if (this._refHasChanged) {
+        this._refHasChanges = false;
+        this._propsAnimated.setNativeView(this._component);
+      }
     }
 
+    _setComponentRef = c => {
+      if (c !== this._component) {
+        this._component = c;
+        this._refHasChanged = true;
+      }
+    };
+
     render() {
       const props = this._propsAnimated.__getProps();
       return (
-        <Component
-          {...props}
-          ref={ref => (this._component = ref)}
-          collapsable={false}
-        />
+        <Component {...props} ref={this._setComponentRef} collapsable={false} />
```

Comment:
```
filterNonAnimatedProps(inputProps) {
  const props = {};
  for (const key in inputProps) {
    const value = inputProps[key];
    if (value instanceof AnimatedNode) {
      if (value instanceof AnimatedStyle) {
        props[key] = this.filterNonAnimatedProps(value);
      }
    } else {
      props[key] = value;
    }
  }
  return props;
}
...
const props = this.filterNonAnimatedProps();
```

Reply:
@kmagiera 
You're right up to a point of implementation. Please concern that in this situation `props` is only a nested map so it has called for another solution 😄